### ISSUE TITLE
Commands will retry and potentially exceed their timeout values during page transition

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#594]
- Cypress Version: 0.19.4

### Is this a Feature or Bug?
Bug

### Current behavior:
If a page transition event occurs whilst a command is currently running + retrying, it will continue to retry during this phase and not only can it unexpected time out, but it can also potentially access properties on the `window` or `document` that can cause DOMException errors.

### Desired behavior:
When Cypress detects a page transition it will pause executing new commands until page stability has been reached.

However currently running commands should have their retry behavior blocked and their internal retry counter reset once page transition is finished and page stability has resumed.

### How to reproduce:

```js
// if the click event on the <a> is async
// then the page transition event will occur
// once the cy.url command begins running

$("a").on("click", () => {
  setTimeout(() => {
    window.location.href = "/new-page"
  }, 100)
})

cy.get("a").click()
cy.url().should("include", "/new-page")
```

**Comment by brian-mann** [comment]
The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
